app/models/game.rb

```diff
@@ -54,14 +54,17 @@ def get_piece_at_coor(x, y)
   end
 
   def under_attack?(is_white, x, y)
-    pieces.where(is_white: !is_white).where.not(x_position: nil, y_position: nil).find_each do |piece|
+    pieces.where(is_white: !is_white).where.not(x_position: nil, y_position: nil, type: KING).find_each do |piece|
       return true if piece.valid_move?(x, y)
     end
+    if (enemy_king(is_white).x_position - x).abs <= 1 && (enemy_king(is_white).y_position - y).abs <= 1
+      return true
+    end
     false
   end
 
   def attacking_piece(is_white) # is_white is the value of the friendly_king, NOT the attacking_piece
-    pieces.where(is_white: !is_white).where.not(x_position: nil, y_position: nil).find_each do |piece|
+    pieces.where(is_white: !is_white).where.not(x_position: nil, y_position: nil, type: KING).find_each do |piece| # a king cannot be an attacking piece because it would put himself into check
```

Comment:
I don't see why this would be the case. If a king is even in location from where it can attack a square, that means it must have already been able to make a valid move to that square it is currently located, meaning a king at that location was not in check at time it moved there. Question of whether king making a capture by moving to square where it would then be in check is irrelevant for purposes of determining checkmate or stalemate, since the king would not actually have to move in order to capture the enemy king, just be in a position to do so given a move by that player whose king is threatened.

Reply:
Like I mentioned elsewhere, I added this in to circumvent an infinite loop. I'm removing it now.